Update

DuckySoLucky opened the pull request:
## Description

Changes: 
`utils/hypixel.js` - Hypixel Level with Progress, UUID
`stats/hypixelLevel.js` - Hypixel Network Level with Progress
`constants/skills.js` - Added `totalXp` and changed `progress` to 0 if the level of the skill is maxed since the next level experience requirement is unknown
`constants/maro_networth/generators/networthGenerator.js`  - If the calculation is empty, it will not be shown
`constants/llyWeight.js` - Added Lily weight Calculator 
`constants/senitherWeight.js` - Moved Senither Weight to different file and modified little bit
`stats/weight.js` - Added Lily Weight 
`stats/bestiary.js` - Revamped, Shows more information
`stats/slayer.js` - `progress` change (like skills) and added `totalKills`
`stats/milestones.js` - `progress` change (like skills), added `nextRarity` - amount of x actions required to get pet
`stats/talismans.js` - If rarity is not legendary or higher, it will not put enrichment tag

## Type of change
- [x] New feature

## Preview

> Github broke space indentation D;
<details>
   <summary>hypixel.js</summary>
   
   ````json
"hypixelLevel": 345,
   ````
   ````json
"uuid": "fb3d96498a5b4d5b91b763db14b195ad",
"hypixelLevel": 345.6783870374495,
   ````

</details>


<details>
   <summary>weight.js</summary>
   
   ```json
"weight": {
  "total_weight": 22584.026777581483,
  "total_weight_with_overflow": 39654.99620022338,
  "weight": {
    "skills": {
      "farming": {
        "weight": 2200,
        "weight_overflow": 15.023391972939985
      },
      "mining": {
        "weight": 1750,
        "weight_overflow": 1062.3826356077184
      },
      "combat": {
        "weight": 1500,
        "weight_overflow": 4591.561092495839
      },
      "foraging": {
        "weight": 850,
        "weight_overflow": 8.789235549378212
      },
      "fishing": {
        "weight": 2500,
        "weight_overflow": 895.6468425432993
      },
      "enchanting": {
        "weight": 450,
        "weight_overflow": 143.99555778184845
      },
      "alchemy": {
        "weight": 200,
        "weight_overflow": 128.95969478086565
      },
      "taming": {
        "weight": 500,
        "weight_overflow": 2044.1260852789337
      }
    },
    "slayer": {
      "revenant": {
        "weight": 452.8985507246377,
        "weight_overflow": 411.5364848149176
      },
      "tarantula": {
        "weight": 472.14353163361665,
        "weight_overflow": 0.20585515645543268
      },
      "sven": {
        "weight": 509.683995922528,
        "weight_overflow": 0.6884923117028161
      },
      "enderman": {
        "weight": 699.3006993006993,
        "weight_overflow": 234.37939051097038
      }
    },
    "dungeons": {
      "catacombs": {
        "weight": 9500,
        "weight_overflow": 7483.734449720966
      },
      "classes": {
        "healer": {
          "weight": 200,
          "weight_overflow": 4.004290171868912
        },
        "mage": {
          "weight": 200,
          "weight_overflow": 26.317111763103135
        },
        "berserk": {
          "weight": 200,
          "weight_overflow": 3.973045663046771
        },
        "archer": {
          "weight": 200,
          "weight_overflow": 11.686545043086344
        },
        "tank": {
          "weight": 200,
          "weight_overflow": 3.9592214749600014
        }
      }
    }
  }
},

   ```
   
   ```json
"weight": {
  "senither": {
    "total": 39654.98622753743,
    "weight": 22584.026777581483,
    "weight_overflow": 8890.474563324871,
    "skills": {
      "farming": {
        "total": 2215.02339197294,
        "weight": 2200,
        "weight_overflow": 15.023391972939985
      },
      "mining": {
        "total": 2812.3743105667354,
        "weight": 1750,
        "weight_overflow": 1062.3743105667354
      },
      "combat": {
        "total": 6091.5605304046685,
        "weight": 1500,
        "weight_overflow": 4591.5605304046685
      },
      "foraging": {
        "total": 858.7892355493782,
        "weight": 850,
        "weight_overflow": 8.789235549378212
      },
      "fishing": {
        "total": 3395.6468425432995,
        "weight": 2500,
        "weight_overflow": 895.6468425432993
      },
      "enchanting": {
        "total": 593.9955577818484,
        "weight": 450,
        "weight_overflow": 143.99555778184845
      },
      "alchemy": {
        "total": 328.9596947808657,
        "weight": 200,
        "weight_overflow": 128.95969478086565
      },
      "taming": {
        "total": 2544.1249997251352,
        "weight": 500,
        "weight_overflow": 2044.124999725135
      }
    },
    "slayer": {
      "total": 2780.837000375528,
      "weight": 2134.026777581482,
      "weight_overflow": 646.8102227940462,
      "revenant": {
        "total": 864.4350355395553,
        "weight": 452.8985507246377,
        "weight_overflow": 411.5364848149176
      },
      "tarantula": {
        "total": 472.34938679007206,
        "weight": 472.14353163361665,
        "weight_overflow": 0.20585515645543268
      },
      "sven": {
        "total": 510.37248823423084,
        "weight": 509.683995922528,
        "weight_overflow": 0.6884923117028161
      },
      "enderman": {
        "total": 933.6800898116696,
        "weight": 699.3006993006993,
        "weight_overflow": 234.37939051097038
      }
    },
    "dungeons": {
      "total": 18033.67466383703,
      "weight": 10500,
      "weight_overflow": 7533.6746638370305,
      "catacombs": {
        "total": 16983.734449720967,
        "weight": 9500,
        "weight_overflow": 7483.734449720966
      },
      "classes": {
        "healer": {
          "total": 204.00429017186892,
          "weight": 200,
          "weight_overflow": 4.004290171868912
        },
        "mage": {
          "total": 226.31711176310313,
          "weight": 200,
          "weight_overflow": 26.317111763103135
        },
        "berserk": {
          "total": 203.97304566304678,
          "weight": 200,
          "weight_overflow": 3.973045663046771
        },
        "archer": {
          "total": 211.68654504308634,
          "weight": 200,
          "weight_overflow": 11.686545043086344
        },
        "tank": {
          "total": 203.95922147496,
          "weight": 200,
          "weight_overflow": 3.9592214749600014
        }
      }
    }
  },
  "lily": {
    "total": 54556.63917155564,
    "skills": {
      "total": 24774.12876497527,
      "base": 19500.51710817874,
      "overflow": 5273.61165679653
    },
    "slayer": {
      "total": 5477.2127194981
    },
    "catacombs": {
      "total": 24305.297687082268,
      "completion": {
        "base": 4424.393109746216,
        "master": 6789.0872632273085
      },
      "experience": 13091.817314108743
    }
  }
}
```
</details>

<details>
   <summary>talismans.js</summary>
   
   ```json
{
  "name": "Dante Talisman",
  "id": "DANTE_TALISMAN",
  "reforge": "None",
  "rarity": "UNCOMMON",
  "recombobulated": true,
  "enrichment": "None"
}
   ```

   ```json
{
  "name": "Dante Talisman",
  "id": "DANTE_TALISMAN",
  "reforge": "None",
  "rarity": "UNCOMMON",
  "recombobulated": true,
}
   ```
</details>


<details>
   <summary>slayer.js</summary>
   
   ```json
"zombie": {
  "xp": 3163500,
  "level": 9,
  "xpForNext": 0,
  "progress": 1,
  "kills": {
    "1": 20,
    "2": 20,
    "3": 154,
    "4": 4608,
    "5": 449
  }
}
   ```

   ```json
"zombie": {
  "xp": 3163500,
  "totalKills": 5251,
  "level": 9,
  "xpForNext": 0,
  "progress": 0,
  "kills": {
    "1": 20,
    "2": 20,
    "3": 154,
    "4": 4608,
    "5": 449
  }
}
   ```
</details>

<details>
   <summary>milestones.js</summary>
   
   ```json
"milestones": {
  "fishing": {
    "sea_creatures_killed": 159569,
    "current_pet": "legendary",
    "next_pet": 0,
    "progress": 1
  },
  "mining": {
    "ores_mined": 3049605,
    "current_pet": "legendary",
    "next_pet": 0,
    "progress": 1
  }
}
   ```

   ```json
"milestones": {
  "fishing": {
    "current_pet": "legendary",
    "next_pet": 0,
    "sea_creatures_killed": 159569,
    "nextRarity": 0,
    "progress": 0
  },
  "mining": {
    "current_pet": "legendary",
    "next_pet": 0,
    "ores_mined": 3049781,
    "nextRarity": 0,
    "progress": 0
  }
}
   ```
</details>

<details>
   <summary>bestiary.js</summary>
   
   ```json
"zombie": {
  "kills": 15146,
  "tier": 5,
  "nextTier": null
},
"arachne_brood": {
  "kills": 17163,
  "tier": 9,
  "nextTier": 7837
}
   ```

   ```json
"zombie": {
  "tier": 5,
  "nextTierKills": null,
  "currentKills": 15146,
  "killsForNext": null,
  "progress": 0,
  "levelWithProgress": 5
},
"arachne_brood": {
  "tier": 9,
  "nextTierKills": 25000,
  "currentKills": 17163,
  "killsForNext": 7837,
  "progress": 0.68652,
  "levelWithProgress": 9.68652
}
   ```
</details>

<details>
   <summary>skills.js</summary>
   
   ```json
"foraging": {
    "xp": 55172425,
    "level": 50,
    "xpCurrent": 2451984,
    "xpForNext": 0,
    "progress": 1,
    "levelWithProgress": 50
  }
   ```

   ```json
"foraging": {
    "totalXp": 57624409.17559126,
    "xp": 55172425,
    "level": 50,
    "xpCurrent": 2451984,
    "xpForNext": 0,
    "progress": 0,
    "levelWithProgress": 50
},
   ```
</details>

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

Altpapier commented:
For weight stuff that must (somehow) be changed I dont want other peoples code to break from any changes made

Altpapier commented on constants/maro_networth/generators/networthGenerator.js:
```diff
@@ -16,6 +16,23 @@ const getNetworth = async function (data, profile, bank) {
         for (const item of data[key].filter((i) => i.price)) {
             category.total += item.price;
+
+            // Looks pretty bad, I know.. if (item?.modified?.calculation?.length > 0) calculation: item.modified.calculation
```
is it needed to not show the calculation if it doesn't exist?

DuckySoLucky replied:
Not really

Altpapier commented on stats/bestiary.js:
```diff
@@ -18,23 +18,33 @@ module.exports = (profile) => {
             result.categories[family][mobName] = {
-                kills: kills || 0,
-                tier: tier || 0,
-                nextTier: toTier,
```
idk if that also breaks code for people nobody probably uses bestiary anyway tho

Altpapier commented on stats/talismans.js:
```diff
@@ -18,20 +18,31 @@ module.exports = async (profile) => {
+                        rarity: getRarity(talisman.tag?.display.Lore).toUpperCase(),
+                        recombobulated: isRecombed,
+                        enrichment: talisman.tag?.ExtraAttributes?.talisman_enrichment ?? 'None'
+                    
```
idk why but this empty line annoys me lmao

Altpapier commented on constants/lilyWeight.js:
```diff
@@ -0,0 +1,966 @@
+module.exports = {
```
If you add lily weight please use the lilyweight npm package (https://www.npmjs.com/package/lilyweight) its easier to update

DuckySoLucky commented:
Pushed new change, added old weight system. Kept new one too (looks kinda messy tho)
Reverted change for networth generator and changed bestiary so it works with previous versions

DuckySoLucky commented:
## Description

Revamped Pets part in API

## Type of change
- [x] New feature

## Preview

<details>
   <summary>pets.js</summary>
   
   ````json
      {
        "name": "[Lvl 100] Enderman ✦",
        "level": 100,
        "xpCurrent": 37598115.449759774,
        "xpForNext": 0,
        "progress": 0,
        "tier": "MYTHIC",
        "type": "ENDERMAN",
        "exp": 39484815.449759774,
        "skin": "ENDERMAN_SLAYER"
      },
   ````
   ````json
      {
        "type": "ENDERMAN",
        "exp": 39484815.449759774,
        "active": false,
        "tier": "MYTHIC",
        "heldItem": "PET_ITEM_COMBAT_SKILL_BOOST_EPIC",
        "candyUsed": 0,
        "skin": "ENDERMAN_SLAYER",
        "price": 42800000,
        "modified": {
          "name": "[Lvl 100] Mythic Enderman ✦",
          "isPet": true
        },
        "rarity": "mythic",
        "stats": {
          "crit_damage": 75
        },
        "texture_path": "/head/8fff41e1afc597b14f77b8e44e2a134dabe161a1526ade80e6290f2df331dc11",
        "name": "[Lvl 100] Enderman ✦",
        "display_name": "Enderman ✦",
        "xpMaxLevel": 25353230,
        "level": 100,
        "xpCurrent": null,
        "xpForNext": 0,
        "progress": 0,
        "lore": [
          "§8Combat Pet, Void Conqueror Skin",
          "",
          "§7Crit Damage: §a+75",
          " ",
          "§6Enderian",
          "§7Take §a30% §7less damage from end monsters",
          " ",
          "§6Teleport Savvy",
          "§7Buffs the Aspect of the End ability granting §a50 §7weapon damage for 5s on use.",
          " ",
          "§6Zealot Madness",
          "§7Increases your odds to find a special Zealot by §a25%.",
          " ",
          "§6Enderman Slayer",
          "§7Gain +§a40% §7more combat xp from endermen",
          "",
          "§6Held Item: §5Combat Exp Boost",
          "§7Gives +§a50% §7pet exp for Combat",
          " ",
          "§bMAX LEVEL",
          "",
          "§7Total XP: §e39.4M §6/ §e25.3M §6(155%)",
          "§7Candy Used: §e0 §6/ §e10",
          ""
        ]
      },
   ````

</details>

## Checklist
- [x] My code does not remove any previously added features
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

DuckySoLucky commented:
## Description

Added `gamemode`, `first_join`, `armor`, `equipment` tag
Deleted `constants/lilyWeight.js` and `constants/pet_items.js`
Added new Endpoints `/v1/items/:user` and `/v1/items/:user/:profile`, returns all inventories which are in Hypixel's API decoded and formatted
Updated Documentation and examples of responses

## Type of change
- [x] New feature

## Preview

<details>
   <summary>hypixel.js</summary>
   
   ````json
   "gamemode": "normal",
   "first_join": 1572965243222,
   ````
</details>

<details>
   <summary>armor.js</summary>
   
   ````json
    "armor": {
      "helmet": {
        "id": 301,
        "Count": 1,
        "tag": {
          "ench": [
            {
              "lvl": 3,
              "id": 8
            }
          ],
          "Unbreakable": 1,
          "HideFlags": 254,
          "display": {
            "Lore": [
              "§7Gear Score: §d687 §8(3732)",
              "§7Strength: §c+23 §9(+12) §8(+135.3)",
              "§7Crit Chance: §c+14.2% §9(+12%) §8(+21%)",
              "§7Crit Damage: §c+23% §9(+12%) §8(+135.3%)",
              "§7Bonus Attack Speed: §c+5% §9(+5%) §8(+7.5%)",
              "§7Health: §a+228 §e(+40) §9(+10) §8(+1,353)",
              "§7Defense: §a+169 §e(+20) §9(+10) §8(+971.7)",
              "§7Speed: §a+10.3 §9(+1) §8(+61.5)",
              "§7Intelligence: §a+39.5 §9(+12) §8(+227.55)",
              " §8[§7⚔§8]",
              "",
              "§d§l§d§lLegion V§9, §9Depth Strider III§9, §9Feather Falling X",
              "§9Growth VI§9, §9Protection VI§9, §9Rejuvenate V",
              "§9Sugar Rush III",
              "",
              "§a◆ Snake Rune III",
              "",
              "§6Full Set Bonus: Superior Blood",
              "§7Most of your stats are increased",
              "§7by §a5%§7 and §6Aspect of the",
              "§6Dragons§7 ability deals §a50%§7",
              "§7more damage.",
              "",
              "§9Renowned Bonus",
              "§7Increases most stats by §a+1%§7.",
              "",
              "§8§l* §8Co-op Soulbound §8§l*",
              "§d§l§ka§r §d§l§d§lMYTHIC DUNGEON BOOTS §d§l§ka"
            ],
            "color": 15883544,
            "Name": "§dRenowned Superior Dragon Boots §6✪§6✪§6✪§6✪§6✪"
          },
          "ExtraAttributes": {
            "color": "242:93:24",
            "runes": {
              "SNAKE": 3
            },
            "dungeon_item_level": 5,
            "originTag": "CRAFTING_GRID_COLLECT",
            "enchantments": {
              "depth_strider": 3,
              "sugar_rush": 3,
              "ultimate_legion": 5,
              "feather_falling": 10,
              "rejuvenate": 5,
              "growth": 6,
              "protection": 6
            },
            "uuid": "06ef18d7-fc92-4ac0-a185-4a6b35e4725c",
            "anvil_uses": 13,
            "hot_potato_count": 10,
            "gems": {
              "UNIVERSAL_0": "FINE",
              "UNIVERSAL_0_gem": "TOPAZ"
            },
            "id": "SUPERIOR_DRAGON_BOOTS",
            "timestamp": 1593447720,
            "recombobulated": true,
            "reforge": "renowned",
            "soulbond": true
          }
        },
        "Damage": 0
      },
      "chestplate": {
        "id": 300,
        "Count": 1,
        "tag": {
          "Unbreakable": 1,
          "HideFlags": 254,
          "display": {
            "Lore": [
              "§7Gear Score: §d853 §8(4756)",
              "§7Strength: §c+23 §9(+12) §8(+135.3)",
              "§7Crit Chance: §c+14.2% §9(+12%) §8(+21%)",
              "§7Crit Damage: §c+23% §9(+12%) §8(+135.3%)",
              "§7Bonus Attack Speed: §c+5% §9(+5%) §8(+7.5%)",
              "§7Health: §a+283 §e(+40) §9(+10) §8(+1,660.5)",
              "§7Defense: §a+235 §e(+20) §9(+10) §8(+1,340.7)",
              "§7Speed: §a+4.3 §9(+1) §8(+24.6)",
              "§7Intelligence: §a+39.5 §9(+12) §8(+227.55)",
              " §8[§7⚔§8]",
              "",
              "§d§l§d§lLegion V§9, §9Growth VI§9, §9Protection VI",
              "§9Rejuvenate V",
              "",
              "§6Full Set Bonus: Superior Blood",
              "§7Most of your stats are increased",
              "§7by §a5%§7 and §6Aspect of the",
              "§6Dragons§7 ability deals §a50%§7",
              "§7more damage.",
              "",
              "§9Renowned Bonus",
              "§7Increases most stats by §a+1%§7.",
              "",
              "§8§l* §8Co-op Soulbound §8§l*",
              "§d§l§ka§r §d§l§d§lMYTHIC DUNGEON LEGGINGS §d§l§ka"
            ],
            "color": 15916817,
            "Name": "§dRenowned Superior Dragon Leggings §6✪§6✪§6✪§6✪§6✪"
          },
          "ExtraAttributes": {
            "color": "242:223:17",
            "dungeon_item_level": 5,
            "originTag": "QUICK_CRAFTING",
            "enchantments": {
              "ultimate_legion": 5,
              "rejuvenate": 5,
              "growth": 6,
              "protection": 6
            },
            "uuid": "73a5ceff-0cdc-453c-81c4-3de61d6b25ab",
            "anvil_uses": 13,
            "hot_potato_count": 10,
            "gems": {
              "UNIVERSAL_0": "FINE",
              "UNIVERSAL_0_gem": "TOPAZ"
            },
            "id": "SUPERIOR_DRAGON_LEGGINGS",
            "timestamp": 1590975360,
            "recombobulated": true,
            "reforge": "renowned",
            "soulbond": true
          }
        },
        "Damage": 0
      },
      "leggings": {
        "id": 299,
        "Count": 1,
        "tag": {
          "Unbreakable": 1,
          "HideFlags": 254,
          "display": {
            "Lore": [
              "§7Gear Score: §d921 §8(5000)",
              "§7Strength: §c+23 §9(+12) §8(+135.3)",
              "§7Crit Chance: §c+14.2% §9(+12%) §8(+21%)",
              "§7Crit Damage: §c+23% §9(+12%) §8(+135.3%)",
              "§7Bonus Attack Speed: §c+5% §9(+5%) §8(+7.5%)",
              "§7Health: §a+305 §e(+40) §9(+10) §8(+1,783.5)",
              "§7Defense: §a+257 §e(+20) §9(+10) §8(+1,463.7)",
              "§7Speed: §a+4.3 §9(+1) §8(+24.6)",
              "§7Intelligence: §a+39.5 §9(+12) §8(+227.55)",
              "§7True Defense: §a+5 §8(+7.5)",
              " §8[§7⚔§8]",
              "",
              "§d§l§d§lLegion V§9, §9Growth VI§9, §9Protection VI",
              "§9Rejuvenate V§9, §9True Protection I",
              "",
              "§6Full Set Bonus: Superior Blood",
              "§7Most of your stats are increased",
              "§7by §a5%§7 and §6Aspect of the",
              "§6Dragons§7 ability deals §a50%§7",
              "§7more damage.",
              "",
              "§9Renowned Bonus",
              "§7Increases most stats by §a+1%§7.",
              "",
              "§8§l* §8Co-op Soulbound §8§l*",
              "§d§l§ka§r §d§l§d§lMYTHIC DUNGEON CHESTPLATE §d§l§ka"
            ],
            "color": 15916817,
            "Name": "§dRenowned Superior Dragon Chestplate §6✪§6✪§6✪§6✪§6✪"
          },
          "ExtraAttributes": {
            "color": "242:223:17",
            "dungeon_item_level": 5,
            "originTag": "QUICK_CRAFTING",
            "enchantments": {
              "true_protection": 1,
              "ultimate_legion": 5,
              "rejuvenate": 5,
              "growth": 6,
              "protection": 6
            },
            "uuid": "cc9b3c8e-5a08-41ea-baca-1269e6e1aab3",
            "anvil_uses": 13,
            "hot_potato_count": 10,
            "gems": {
              "UNIVERSAL_0": "FINE",
              "UNIVERSAL_0_gem": "TOPAZ"
            },
            "id": "SUPERIOR_DRAGON_CHESTPLATE",
            "timestamp": 1590975360,
            "recombobulated": true,
            "reforge": "renowned",
            "soulbond": true
          }
        },
        "Damage": 0
      },
      "boots": {
        "id": 397,
        "Count": 1,
        "tag": {
          "ench": [
            {
              "lvl": 3,
              "id": 5
            },
            {
              "lvl": 1,
              "id": 6
            }
          ],
          "HideFlags": 254,
          "SkullOwner": {
            "Id": "f69ba621-a8b6-31a7-8de1-dc7ade140e1d",
            "Properties": {
              "textures": [
                {
                  "Value": "eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHBzOi8vdGV4dHVyZXMubWluZWNyYWZ0Lm5ldC90ZXh0dXJlLzFlMDMyOWY0MjE5MmVlN2MxYTBjNzA0ZjgyZGJiYmU3YzAwZmJmYTNmMDIwYzEwNjdhMjA4NjMwYjk5MWI5ODgifX19"
                }
              ]
            }
          },
          "display": {
            "Lore": [
              "§7Gear Score: §d722 §8(4019)",
              "§7Strength: §c+23 §9(+12) §8(+135.3)",
              "§7Crit Chance: §c+14.2% §9(+12%) §8(+21%)",
              "§7Crit Damage: §c+23% §9(+12%) §8(+135.3%)",
              "§7Bonus Attack Speed: §c+5% §9(+5%) §8(+7.5%)",
              "§7Health: §a+239 §e(+40) §9(+10) §8(+1,414.5)",
              "§7Defense: §a+191 §e(+20) §9(+10) §8(+1,094.7)",
              "§7Speed: §a+4.3 §9(+1) §8(+24.6)",
              "§7Intelligence: §a+39.5 §9(+12) §8(+227.55)",
              " §8[§7⚔§8]",
              "",
              "§d§l§d§lLegion V§9, §9Aqua Affinity I§9, §9Growth VI",
              "§9Protection VI§9, §9Rejuvenate V§9, §9Respiration III",
              "",
              "§6Full Set Bonus: Superior Blood",
              "§7Most of your stats are increased",
              "§7by §a5%§7 and §6Aspect of the",
              "§6Dragons§7 ability deals §a50%§7",
              "§7more damage.",
              "",
              "§9Renowned Bonus",
              "§7Increases most stats by §a+1%§7.",
              "",
              "§8§l* §8Co-op Soulbound §8§l*",
              "§d§l§ka§r §d§l§d§lMYTHIC DUNGEON HELMET §d§l§ka"
            ],
            "Name": "§dRenowned Superior Dragon Helmet §6✪§6✪§6✪§6✪§6✪"
          },
          "ExtraAttributes": {
            "bossId": "b65dd5c4-1a62-409e-a696-238a84631f90",
            "dungeon_item_level": 5,
            "originTag": "BOSS_DROP",
            "enchantments": {
              "ultimate_legion": 5,
              "rejuvenate": 5,
              "respiration": 3,
              "growth": 6,
              "protection": 6,
              "aqua_affinity": 1
            },
            "uuid": "0f19cbe6-6da1-4ee7-a3b6-3dd45033803e",
            "anvil_uses": 13,
            "hot_potato_count": 10,
            "gems": {
              "UNIVERSAL_0": "FINE",
              "UNIVERSAL_0_gem": "TOPAZ"
            },
            "spawnedFor": "1bf1249e-1143-4234-9ddc-77d21d84bf01",
            "id": "SUPERIOR_DRAGON_HELMET",
            "timestamp": 1589287440,
            "recombobulated": true,
            "reforge": "renowned",
            "soulbond": true
          }
        },
        "Damage": 3
      }
    },
   ````
</details>

<details>
   <summary>equipment.js</summary>
   
   ````json
   "equipment": {
      "necklace": {
        "id": 397,
        "Count": 1,
        "tag": {
          "HideFlags": 254,
          "SkullOwner": {
            "Id": "fcb2fa38-e4f6-3ebc-a530-0ece534c1e39",
            "Properties": {
              "textures": [
                {
                  "Value": "ewogICJ0aW1lc3RhbXAiIDogMTY0MzU5OTM1NjU3OCwKICAicHJvZmlsZUlkIiA6ICI0NmY3N2NjNmQ2MjU0NjEzYjc2NmYyZDRmMDM2MzZhNiIsCiAgInByb2ZpbGVOYW1lIiA6ICJNaXNzV29sZiIsCiAgInNpZ25hdHVyZVJlcXVpcmVkIiA6IHRydWUsCiAgInRleHR1cmVzIiA6IHsKICAgICJTS0lOIiA6IHsKICAgICAgInVybCIgOiAiaHR0cDovL3RleHR1cmVzLm1pbmVjcmFmdC5uZXQvdGV4dHVyZS9kNGNiZTg5M2UwZjgyMzAxYzVjZGNlMWI4MTgxODAzYjRmYjgzY2JmMzAzMDRkYjEzOWEzM2Q1NjlkNGVkOGMiLAogICAgICAibWV0YWRhdGEiIDogewogICAgICAgICJtb2RlbCIgOiAic2xpbSIKICAgICAgfQogICAgfQogIH0KfQ=="
                }
              ]
            }
          },
          "display": {
            "Lore": [
              "§7Strength: §c+6 §9(+5)",
              "§7Crit Damage: §c+1%",
              "§7Sea Creature Chance: §c+1%",
              "§7Health: §a+1",
              "§7Defense: §a+7 §9(+6)",
              "§7Speed: §a+1",
              "§7Intelligence: §a+1",
              "§7Magic Find: §a+1",
              "§7Ferocity: §a+1",
              "",
              "§bArachno Resistance I",
              "§7Grants §a+1.5❈ Defense §7against",
              "§7§a§aspiders§7.",
              "§bLifeline I",
              "§7When under §c20% §7maximum HP,",
              "§7your §c❁ Damage §7and §a❈",
              "§aDefense §7increase by §a2.5%§7.",
              "",
              "§7§7Increase the fire damage you",
              "§7receive by §c8% §7per §c+1❁",
              "§cStrength§7.",
              "",
              "§8§l* §8Co-op Soulbound §8§l*",
              "§5§lEPIC NECKLACE"
            ],
            "Name": "§5Strengthened Delirium Necklace"
          },
          "ExtraAttributes": {
            "attributes": {
              "lifeline": 1,
              "arachno_resistance": 1
            },
            "id": "DELIRIUM_NECKLACE",
            "uuid": "8ca339e3-6161-410e-80e1-8e04f3cc10c7",
            "timestamp": 1651337520,
            "reforge": "strengthened",
            "soulbond": true
          }
        },
        "Damage": 3
      },
      "cloak": {
        "id": 397,
        "Count": 1,
        "tag": {
          "HideFlags": 254,
          "SkullOwner": {
            "Id": "e7b0b792-7786-3b14-82b0-d04f7b5fbf7c",
            "Properties": {
              "textures": [
                {
                  "Value": "ewogICJ0aW1lc3RhbXAiIDogMTY0NzAxNjQzNDExNiwKICAicHJvZmlsZUlkIiA6ICJlZThjNWMzMGY3NWU0N2QxOTBmOTllNjI5NDgyOGZjMSIsCiAgInByb2ZpbGVOYW1lIiA6ICJTcGFya19QaGFudG9tIiwKICAic2lnbmF0dXJlUmVxdWlyZWQiIDogdHJ1ZSwKICAidGV4dHVyZXMiIDogewogICAgIlNLSU4iIDogewogICAgICAidXJsIiA6ICJodHRwOi8vdGV4dHVyZXMubWluZWNyYWZ0Lm5ldC90ZXh0dXJlLzZjMThiYjBmNTJmMjQ2ZWM5YmZmNWYwNTc2MTQ1NmJlMjczZTczMGNkMjc2NTU1ZmQ3NzJiYThjM2ZhMGYxNDQiCiAgICB9CiAgfQp9"
                }
              ]
            }
          },
          "display": {
            "Lore": [
              "§7Strength: §c+20",
              "§7Health: §a+30",
              "§7Defense: §a+20",
              "§7Intelligence: §a+6 §9(+6)",
              "§7Mining Fortune: §a+12 §9(+12)",
              "",
              "§bBreeze VI",
              "§7When you haven't been damaged in",
              "§7the last §a5s §7or more, your",
              "§7mana costs are decreased by",
              "§7§a6%§7.",
              "§bMana Pool V",
              "§7Increases your maximum mana by",
              "§7§b+100§7.",
              "",
              "§8Tiered Bonus: Molten Core (1/4)",
              "§7Gain §a5% §7damage resistance",
              "§7while fighting §6Kuudra§7.",
              "",
              "§6§l§ka§r §6§l§6§lLEGENDARY CLOAK §6§l§ka"
            ],
            "Name": "§6Glistening Molten Cloak"
          },
          "ExtraAttributes": {
            "attributes": {
              "breeze": 6,
              "mana_pool": 5
            },
            "id": "MOLTEN_CLOAK",
            "boss_tier": 1,
            "uuid": "8ba57c05-8d13-4103-982d-e9a0bb113d11",
            "timestamp": 1657234020,
            "recombobulated": true,
            "reforge": "glistening"
          }
        },
        "Damage": 3
      },
      "belt": {
        "id": 397,
        "Count": 1,
        "tag": {
          "HideFlags": 254,
          "SkullOwner": {
            "Id": "787b3f09-a513-332c-ada2-558ea2c77575",
            "Properties": {
              "textures": [
                {
                  "Value": "ewogICJ0aW1lc3RhbXAiIDogMTY0MzU5OTU2NjgwMCwKICAicHJvZmlsZUlkIiA6ICIwZjczMDA3NjEyNGU0NGM3YWYxMTE1NDY5YzQ5OTY3OSIsCiAgInByb2ZpbGVOYW1lIiA6ICJPcmVfTWluZXIxMjMiLAogICJzaWduYXR1cmVSZXF1aXJlZCIgOiB0cnVlLAogICJ0ZXh0dXJlcyIgOiB7CiAgICAiU0tJTiIgOiB7CiAgICAgICJ1cmwiIDogImh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYzNiMGQ5NjIyZGY2OTE1OWMyMDcwNmU5YWY0MDBkNmJiMzA4ZWJjYjc1MjAwMjVmNTdkOTc3YWQwMzRmNmI1YSIsCiAgICAgICJtZXRhZGF0YSIgOiB7CiAgICAgICAgIm1vZGVsIiA6ICJzbGltIgogICAgICB9CiAgICB9CiAgfQp9"
                }
              ]
            }
          },
          "display": {
            "Lore": [
              "§7Health: §a+35",
              "§7Defense: §a+25",
              "",
              "§7Reduces damage taken by §c6%§7.",
              "",
              "§7§8This item can be reforged!",
              "§8§l* §8Soulbound §8§l*",
              "§d§l§ka§r §d§l§d§lMYTHIC BELT §d§l§ka"
            ],
            "Name": "§dBlack Belt"
          },
          "ExtraAttributes": {
            "id": "DOJO_BLACK_BELT",
            "uuid": "08eaf4b5-44b8-4dc0-89ae-890d1cb3833b",
            "timestamp": 1650471720,
            "recombobulated": true,
            "soulbond": true
          }
        },
        "Damage": 3
      },
      "gloves": {
        "id": 397,
        "Count": 1,
        "tag": {
          "HideFlags": 254,
          "SkullOwner": {
            "Id": "5aa6d1ce-ce5d-3695-99f7-4aeb5b77c956",
            "Properties": {
              "textures": [
                {
                  "Value": "ewogICJ0aW1lc3RhbXAiIDogMTY1MDM3OTExMjUxMywKICAicHJvZmlsZUlkIiA6ICJkOGNkMTNjZGRmNGU0Y2IzODJmYWZiYWIwOGIyNzQ4OSIsCiAgInByb2ZpbGVOYW1lIiA6ICJaYWNoeVphY2giLAogICJzaWduYXR1cmVSZXF1aXJlZCIgOiB0cnVlLAogICJ0ZXh0dXJlcyIgOiB7CiAgICAiU0tJTiIgOiB7CiAgICAgICJ1cmwiIDogImh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvMTY4YTJiZmU2ZTU2ZGRlYTEwM2RlOGNlYzAwMGVmYTgzZWJiOWE5NTljNjhkODI0MDI1NDNjMDFkNDEzN2QyNiIsCiAgICAgICJtZXRhZGF0YSIgOiB7CiAgICAgICAgIm1vZGVsIiA6ICJzbGltIgogICAgICB9CiAgICB9CiAgfQp9"
                }
              ]
            }
          },
          "display": {
            "Lore": [
              "§7Strength: §c+15 §9(+5)",
              "§7Crit Damage: §c+30%",
              "§7Sea Creature Chance: §c+1.5%",
              "§7Health: §a+35",
              "§7Defense: §a+6 §9(+6)",
              "§7Ferocity: §a+2",
              "",
              "§8Tiered Bonus: Fireproof (1/2)",
              "§7Provides immunity to §6nether",
              "§6magma§7.",
              "",
              "§8Tiered Bonus: Lord's Blessing (1/5)",
              "§7Grants §3+1α Sea Creature",
              "§3Chance§7.",
              "§7Grants §b+5✯ Magic Find §7on",
              "§7Lava Sea Creatures.",
              "",
              "§7Damage taken by §cLava Sea",
              "§cCreatures §7is reduced by §a5%",
              "§a§7and damage dealt is increased",
              "§7by §c1.1x",
              "",
              "§8§l* §8Co-op Soulbound §8§l*",
              "§5§lEPIC GLOVES"
            ],
            "Name": "§5Strengthened Magma Lord Gauntlet"
          },
          "ExtraAttributes": {
            "id": "MAGMA_LORD_GAUNTLET",
            "uuid": "4c009dbb-d2fe-4833-a65b-df56cec27037",
            "timestamp": 1651102980,
            "reforge": "strengthened",
            "soulbond": true
          }
        },
        "Damage": 3
      }
    },
   ````
</details>

## Checklist
- [x] My code does not remove any previously added features
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings
